### PR TITLE
Refactor S2ST

### PR DESCRIPTION
**Pull request by hirofumi0810**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [x] Did you read the [contributor guideline](https://github.com/pytorch/fairseq/blob/main/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?
Moved sub-modules used for S2ST models to separate files. I also made minor fixes specific to speech-to-text/speech translation tasks.

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: fairseq/models/speech_to_text/berard.py [hunk 1]
```diff
@@ -132,7 +132,7 @@ def build_encoder(cls, args, task):
             lstm_size=args.lstm_size,
             dropout=args.dropout,
         )
-        if getattr(args, "load_pretrained_encoder_from", None):
+        if getattr(args, "load_pretrained_encoder_from", None) is not None:
```
Comment (sravyapopuri388) [inline fairseq/models/speech_to_text/berard.py:135]: nit: is not None condition check is not needed I guess?
```suggestion
        if getattr(args, "load_pretrained_encoder_from", None):
```

File: fairseq/models/speech_to_text/berard.py [hunk 2]
```diff
@@ -150,7 +150,7 @@ def build_decoder(cls, args, task):
             attention_dim=args.attention_dim,
             output_layer_dim=args.output_layer_dim,
         )
-        if getattr(args, "load_pretrained_decoder_from", None):
+        if getattr(args, "load_pretrained_decoder_from", None) is not None:
```
Comment (sravyapopuri388) [inline fairseq/models/speech_to_text/berard.py:153]: nit: is not None condition check is not needed I guess?
```suggestion
        if getattr(args, "load_pretrained_encoder_from", None):
```

File: fairseq/models/speech_to_text/convtransformer.py [hunk 3]
```diff
@@ -162,7 +162,7 @@ def add_args(parser):
     @classmethod
     def build_encoder(cls, args):
         encoder = ConvTransformerEncoder(args)
-        if getattr(args, "load_pretrained_encoder_from", None):
+        if getattr(args, "load_pretrained_encoder_from", None) is not None:
```
Comment (sravyapopuri388) [inline fairseq/models/speech_to_text/convtransformer.py:165]: nit: is not None condition check is not needed I guess?
```suggestion
        if getattr(args, "load_pretrained_encoder_from", None):
```

File: fairseq/models/speech_to_text/convtransformer.py [hunk 4]
```diff
@@ -171,7 +171,7 @@ def build_encoder(cls, args):
     @classmethod
     def build_decoder(cls, args, task, embed_tokens):
         decoder = TransformerDecoderNoExtra(args, task.target_dictionary, embed_tokens)
-        if getattr(args, "load_pretrained_decoder_from", None):
+        if getattr(args, "load_pretrained_decoder_from", None) is not None:
```
Comment (sravyapopuri388) [inline fairseq/models/speech_to_text/convtransformer.py:174]: nit: is not None condition check is not needed I guess?
```suggestion
        if getattr(args, "load_pretrained_encoder_from", None):
```

File: fairseq/models/speech_to_text/xm_transformer.py [hunk 5]
```diff
@@ -632,7 +632,7 @@ def build_model(cls, args, task):
 
         # make sure all arguments are present in older models
         base_architecture(args)
-        if getattr(args, "load_pretrained_decoder_from", None):
+        if getattr(args, "load_pretrained_decoder_from", None) is not None:
```
Comment (sravyapopuri388) [inline fairseq/models/speech_to_text/xm_transformer.py:642]: nit: is not None condition check is not needed I guess?
```suggestion
        if getattr(args, "load_pretrained_encoder_from", None):
```

**Review by sravyapopuri388 — APPROVED** [review]
Looks good except for nits

File: fairseq/models/speech_to_text/convtransformer.py (hunk 4 above)
Comment (hirofumi0810) [reply fairseq/models/speech_to_text/convtransformer.py:174]: @sravyapopuri388 I just followed Google Python Style Guide.

Comment (hirofumi0810) [reply fairseq/models/speech_to_text/convtransformer.py:174]: @sravyapopuri388 wdyt?

**Pull request merged**